src/bson.rs

```diff
@@ -973,76 +970,78 @@ impl Timestamp {
     }
 }
 
-/// `DateTime` representation in struct for serde serialization
+/// Struct representing a BSON datetime.
+///
+/// Is is reccomended to use a [`chrono::DateTime`] for date operations
+/// and to convert it to/from a [`crate::DateTime`] via the `From`/`Into` implementations.
+///
+/// ```
+/// use chrono::prelude::*;
+/// # fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
+/// let chrono_dt: chrono::DateTime<Utc> = "2014-11-28T12:00:09Z".parse()?;
+/// let bson_dt: bson::DateTime = chrono_dt.into();
+/// let back_to_chrono: chrono::DateTime<Utc> = bson_dt.into();
+/// # Ok(())
+/// # }
+/// ```
 ///
-/// Just a helper for convenience
+/// This type differs from [`chrono::DateTime`] in that it serializes to and deserializes from a
+/// BSON datetime rather than an ISO-8601 formatted string. This means that it will serialize to and
```

Comment:
what does encoding as an extended JSON object entail in a non-BSON, non-extended JSON format? using the extJSON key names?

Reply:
Yep, so the way the crate implemented `Serialize` / `Deserialize` for the various types was to do BSON in BSON and extended JSON (or its equivalent) for all other serialization formats (This is something I suggested we try in Swift a while back). Users have expressed a decent amount of frustration around this (particularly around DateTime), though the serde helpers have helped a lot. I added this comment in here to further help clarify the situation, though I agree that part is a bit confusing. Updated the wording to be a bit more specific.

Given that we're breaking API, we could consider changing this approach, though I think that would probably require a lot of work. We could additively support alternate default serialization formats in the future via feature flags too, so it's not completely necessary to do it now.

Reply:
ah I see. yeah, I don't think we should try to tackle this now given there are ways to work around it, but seems like something good to keep in mind for the future.

Reply:
Filed RUST-801.